app/src/main/java/org/mifos/mobile/viewModels/LoginViewModel.kt

```diff
@@ -0,0 +1,178 @@
+package org.mifos.mobile.viewModels
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.observers.DisposableObserver
+import io.reactivex.plugins.RxJavaPlugins
+import io.reactivex.schedulers.Schedulers
+import org.mifos.mobile.api.BaseApiManager
+import org.mifos.mobile.api.DataManager
+import org.mifos.mobile.api.local.PreferencesHelper
+import org.mifos.mobile.models.Page
+import org.mifos.mobile.models.User
+import org.mifos.mobile.models.client.Client
+import org.mifos.mobile.models.payload.LoginPayload
+import org.mifos.mobile.utils.Constants
+import retrofit2.HttpException
+import java.net.UnknownHostException
+import javax.inject.Inject
+
+class LoginViewModel @Inject constructor(private val dataManager: DataManager?) : ViewModel() {
```

Comment:
In MVVM architecture, don't we have a Repository as well??

Reply:
yes we do and I had a doubt as to how we are fetching the data from the source incase of login activity

Reply:
I'm unsure I understand what you mean by "source". However, I recommend creating a single repository called `UserAuthRepository`, including the logic for login and registration, and then using it in both the Register and Login view models.